src/main/java/com/spotify/logging/LoggingConfigurator.java

```diff
@@ -433,6 +427,15 @@ public static void configure(final File file, final String defaultIdent) {
     StatusPrinter.printInCaseOfErrorsOrWarnings(context);
   }
 
+  private static LoggerContext setupLoggerContext(Logger rootLogger, String ident) {
+    final LoggerContext context = rootLogger.getLoggerContext();
+    context.reset();
+    context.putProperty("ident", ident);
+    context.putProperty("pid", getMyPid());
+    context.putProperty("hostname", getHeliosHostname());
```

Comment:
This seems like it could be problematic, indeed - has it been tested in different circumstances? A different approach could be something like only calling `context.putProperty` if `getHeliosHostname()` returns non-null, ensuring that the default hostname handling will come into play.